src/System.Runtime.Extensions/src/System/IO/Path.Windows.cs

```diff
@@ -107,7 +122,9 @@ public static string GetFullPath(string path)
 
             // Emulate FileIOPermissions checks, retained for compatibility
             CheckInvalidPathChars(fullPath, true);
-            if (fullPath.Length > 2 && fullPath.IndexOf(':', 2) != -1)
+
+            int startIndex = fullPath.StartsWith(LongPathPrefix, StringComparison.Ordinal) ? LongPathPrefix.Length + 2 : 2;
```

Comment:
There are lots of magic numbers floating around here (a bunch of twos so far).  It'd be helpful if these were put into named constants to help describe what they are.